website/blog/2022-10-03-aluminium.md

```diff
@@ -0,0 +1,789 @@
+script is contained within a directory that also contains other sources, that
+script will be treated as a standalone one in order to avoid flaky compilation
+errors coming from normal files in the workspace.
+
+## [Scala 3] Auto import symbols in string interpolation
+
+Previously, the only suggestions for string interpolations were coming from the
+currently available symbols in scope. This meant that if you wanted to import
+something from another package, you would need to do it manually.
+
+This problem is now resolved. Users can easily get such symbols automatically
+imported, which creates a seamless workflow.
+
+In addition to importing standard symbol, we will also suggest importing
+extension methods.
+
+TODO GIf ![auto-import](TODO)
+
+The feature was already available for Scala 2.
+
+## [Scala 3] Add autofill named arguments completion
+
+Thanks to great work by [jkciesluk](https://github.com/jkciesluk) Scala 3
+completions will also show `autofill all` completion item, which will try to
+fill in all the arguments of the current method using the symbol in scope or
+with `???` if no symbols match. Users can later tabulate over the different
+proposal to adjust them.
+
+![auto-import](TODO)
+
+The feature was already available for Scala 2.
+
+## [Scala 3] Improved interactive features performance
+
+Interactive features are the ones that are supposed to work within the current
+file and provide seamless experience even if the code doesn't compile. These
+feature are for example hover, completions, document highlight as well some code
+actions.
+
+Previously, we would recompile the file whenever each of these features were
+invoked. This was causing the compiler usage of CPU spike occasionally and would
+make all of these features slower. Now, the file will only be recompiled if it
+changed, which results in a much faster performance across Scala 3 codebases.
+
+Additionally a regression was found for completions, where would try to
+calculate documentation for all of the completion items despite the fact that it
+can be done lazily. This was not fixed and improved completions performance
+around 40%.
+
+Thanks to [tanishiking](https://github.com/tanishiking)) for investigation and
+his great fixes.
+
+## Make documentHighlight an interactive feature
+
+Document highlight allows users to see all occurrences of the symbol currently
+under the cursor in a single file. This was one of the first
+[features](https://scalameta.org/metals/blog/2019/04/12/mercury#document-highlight)
+developed by me for Metals and initially it was done using semanticdb. However,
+this means that document highlight will only work in a fully compiled workspace
+and will not show new symbols if the code was recompiled. Since the feature only
+works within a single file, this was a perfect candidate for turning into
+interactive feature, which would use the Scala presentation compiler already
+used in hovers or completions.
```

Comment:
I feel it's a bit going into technical details, but I think release notes (of apps) shouldn't go into technical details.
I'm wondering if maybe we can just mention something like "previously, document highlight didn't work perfectly if the file doesn't compile, now it works for in-compilable source file". And optionally mention the details in short like "Previously, it didn't work because ..., but now it works because it uses presentation compiler....".

What do you think?

Reply:
I changed it a bit to make it more user facing.